main_model.py

```diff
@@ -0,0 +1,101 @@
+def detect_document_type(document_path):
+    
+    guess_file = guess(document_path)
+    file_type = ""
+    image_types = ['jpg', 'jpeg', 'png', 'gif']
+    
+    if(guess_file.extension.lower() == "pdf"):
+        file_type = "pdf"
+        
+    elif(guess_file.extension.lower() in image_types):
+        file_type = "image"
+        
+    else:
+        file_type = "unkown"
+        
+    return file_type
+
+from langchain.document_loaders.image import UnstructuredImageLoader
+from langchain.document_loaders import UnstructuredFileLoader
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The imports for `UnstructuredImageLoader` and `UnstructuredFileLoader` should be placed at the top of the file, following Python's best practices for import statements. This improves readability and maintainability.
```diff
- from langchain.document_loaders.image import UnstructuredImageLoader
- from langchain.document_loaders import UnstructuredFileLoader
```
Add these lines at the top of your script.


<!-- This is an auto-generated comment by OSS CodeRabbit -->